cmake/InstallDirs.cmake

```diff
@@ -0,0 +1,41 @@
+if(WIN32)
+  if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    set(program_files "$ENV{ProgramFiles}")
+
+    # For 32 bit builds.
+    if(CMAKE_C_SIZEOF_DATA_PTR EQUAL 4 AND ENV{ProgramFiles\(x86\)})
+      set(program_files "$ENV{ProgramFiles\(x86\)}")
+    endif()
+
+    if(NOT program_files)
+      if(NOT CMAKE_C_SIZEOF_DATA_PTR EQUAL 4)
+        set(program_files "\\Program Files")
+      else()
+        set(program_files "\\Program Files (x86)")
+      endif()
+    endif()
+
+    string(REPLACE "\\" "\\\\" program_files "${program_files}")
```

Comment:
CMake's native separator format is slash, and the last thing you want to have is mixing slashes and backslashes. It's a nightmare.

Always use slashes and at the input and output convert to and from "native" format. I'm using "native" here because Windows can handle happily slashes.

Use [`file(TO_CMAKE_PATH)` ](https://cmake.org/cmake/help/latest/command/file.html?#to-cmake-path) to convert the `$ENV` paths and have less escaping and in the end simpler code.

Reply:
I just needed this for the `config.h` interpolated variable, and not this one, I will use a separate variable for that.

Reply:
Fixed, thanks!